src/common.c

```diff
@@ -413,8 +413,9 @@ const char *ofi_straddr(char *buf, size_t *len,
 		break;
 	case FI_ADDR_PSMX3:
 		size =
-		    snprintf(buf, *len, "fi_addr_psmx3://%" PRIx64 ":%" PRIx64,
-			     *(uint64_t *)addr, *((uint64_t *)addr + 1));
+		    snprintf(buf, *len, "fi_addr_psmx3://%" PRIx64 ":%" PRIx64 ":%" PRIx64 ":%" PRIx64,
+			     *(uint64_t *)addr, *((uint64_t *)addr + 1),
+			     *((uint64_t *)addr + 2), *((uint64_t *)addr + 3));
```

Comment:
This looks like a breaking change to the ABI in the format of the address.  This is not backwards compatible.  FI_ADDR_PSMX3 cannot change like this.  You need some other solution (such as supporting 2 address formats now) and should maintain compatibility.